### PR TITLE
Zadanie 2

### PR DESCRIPTION
**Pull request by higar4**

### CODE REVIEW

**Review by liudmylatkachenkooffcl — COMMENTED** [review]
Ogólnie, jest dobrze, i nie mam szczególnych uwag. Jedynie co, uważam wybór KNN za ciekawy, ponieważ do rozwiązania danego typu problemów rzadko korzysta się z klasteryzacji(było dużo zmiennych categorical albo 0/1). Chociaż, wybrałeś odpowiednie zmienne liczbowe(int/float), co umożliwiło istnienie danego modelu. Do tego, miał naprawdę  wysoką wydajność.

**Review by Justa555 — COMMENTED** [review]
Zadanie spełnia większość wymagań. Nie dopatrzyłam się tylko odpalenia modeli na danych treningowych i wyciągnięcia z tego wniosków. Nie wiem czy tak mogło być, ale z metryk widzę tylko dokładność, a ja poprzez wybór metryk rozumiem więcej niż jedną. Oprócz tego raczej reszta jest git.  Podobają mi się wykresy zależności hiperparametrów od dokładności. Może osobiście dodałabym więcej komentarzy przed wnioskami dla lepszego zrozumienia co się dzieje.

**Review by jakub206 — COMMENTED** [review]
Ogólnie z tego co widzę wszystkie warunki zadania zostały spełnione, poza tym że akurat skupiłeś się tylko na 1 metryce a z polecenia wnioskuję że miało być ich kilka. Komentarze szczegółowe, wnioski są więc gites. Wykresy accuracy od hiperparametrów bardzo fajne, podobają mi się. Fajnie, że też sprawdziłeś szansę przeżycia dla wybranych przez ciebie osób

File: ML_zadanie.ipynb [hunk 1]
```diff
@@ -0,0 +1,843 @@
...
+        "drive.mount('/content/drive')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "import pandas as pd\n",
+        "import numpy as np\n",
+        "from pathlib import Path\n",
+        "base_path = Path('/content/drive/MyDrive/wakacyjne/llm/02') # change path here!"
```
Comment (woznin) [inline ML_zadanie.ipynb:47]: Ciekawa scieżka /llm 😅

File: ML_zadanie.ipynb [hunk 2]
```diff
@@ -0,0 +1,843 @@
...
+        "- Jak nie przetrenować wybranego modelu?\n",
+        "- Jaki wynik klasyfikacji możemy uznać za *dobry*?\n",
+        "\n",
+        "\n",
+        "Wymagania:\n",
+        "- Wypisz obserwacje z pierwszego zadania, które pomogą Ci w tym. Co było przydatne, a co okazało się bezużyteczne?\n",
+        "- [Nie doprowadź](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) do ~~przecieku statku~~ wycieku danych (np. nie ucz modelu na danych testowych). Nauczone modele odpal na danych treningowych i testowych - opisz uzyskane wyniki.\n",
+        "- Stwórz baseline, czyli dla porównania sprawdź jak z zadaniem radzi sobie [Dummy Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.dummy.DummyClassifier.html) (jeśli Twój docelowy model radzi sobie gorzej - uciekaj)\n",
+        "- Przeprowadź badania na dwóch wybranych modelach uczenia maszynowego (np. spośród: drzew decyzyjnych, SVM, MLP, KNN, z gwiazdką [XGBoost](https://xgboost.readthedocs.io/en/stable))\n",
+        "- W badaniach użyj wybranych metryk. Wybór uzasadnij.\n",
```
Comment (woznin) [inline ML_zadanie.ipynb:96]: Tego punktu mi brakuje, mocna skupka na accuracy poszła

File: ML_zadanie.ipynb [hunk 3]
```diff
@@ -0,0 +1,843 @@
...
+        "- Czego nauczyło Cię o badanym zbiorze danych poprzednie zadanie? Jak możesz wykorzystać wyciągnięte z niego wnioski w procesie tworzenia modelu?\n",
+        "- Jak przeprowadzenie standaryzacji danych może wpływać na zachowanie modelu?\n",
+        "- Co mój model robi i w jaki sposób?\n",
+        "- Jak nie przetrenować wybranego modelu?\n",
+        "- Jaki wynik klasyfikacji możemy uznać za *dobry*?\n",
+        "\n",
+        "\n",
+        "Wymagania:\n",
+        "- Wypisz obserwacje z pierwszego zadania, które pomogą Ci w tym. Co było przydatne, a co okazało się bezużyteczne?\n",
+        "- [Nie doprowadź](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) do ~~przecieku statku~~ wycieku danych (np. nie ucz modelu na danych testowych). Nauczone modele odpal na danych treningowych i testowych - opisz uzyskane wyniki.\n",
```
Comment (woznin) [inline ML_zadanie.ipynb:93]: Nauczone modele odpalane były tylko na tescie. Fajnie by było zobaczyć test vs trening, żeby zobaczyc na wlasne oczy czy model przypadkiem nie radzi sobie podejrzanie dobrze na treningowych danych a na testowych leży

File: ML_zadanie.ipynb [hunk 4]
```diff
@@ -0,0 +1,843 @@
...
+            "[[46 59]\n",
+            " [39 35]]\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Możemy zauważyć, że DummyClassifier ma skuteczność na poziomie 0.45 czyli 45% dlatego przy implentowaniu Drzewa Decyzyjnego i KNN będzie mi zależało uzyskać wyższy wynik od 0.45\n",
```
Comment (woznin) [inline ML_zadanie.ipynb:446]: Owszem, ale myśle, ze za baseline możemy wykorzystać coś delikatnie mądrzejszego niż DummyClassifier(strategy='uniform'), np. `strategy="most_frequent"`, wzięcie uniform = rzucanie monetą na każdą próbke, most_frequent symuluje wybieranie zawsze najpopularniejszej klasy więc tutaj automatycznie ten próg by się nam zwiększył.

**Review by woznin — COMMENTED** [review]
Ogólnie zadanie jest zrobione okej: split jest poprawny, dummy jest, dwa modele, siatka hiperparametrów, wykresy i wnioski są, plus fajny dodatek z szansami konkretnych osób. Kilka rzeczy bym doprecyzował, bo trochę psują interpretację wyników.

Największy problem to bardzo pochopny wniosek ze KNN jest lepszy - dosłownie pare komórek wyżej widać, ze na grid searchu drzewa radziły sobie lepiej 🤔 

Oprócz tego zobacz komentarze, fajnie by było zobaczyć proces myślowy idący za tym że accuracy jest wg. Ciebie najważniejszą metryką tutaj. 

PR'a nie musisz poprawiać, ale jako dobrowolna praca domowa zerknij sobie [tutaj](https://medium.com/@aditib259/why-accuracy-isnt-enough-a-guide-to-ai-ml-model-evaluation-f97c0c394375) 🫡

File: ML_zadanie.ipynb (hunk 1 above)
Comment (Dominik-Galus) [reply ML_zadanie.ipynb:47]: image
